magento/magento2
# Greatly improve performance by caching subsequent glob calls

### Description (*)
This pull request introduces an in-memory cache for glob system calls. On systems where file I/O is slow (for example Docker for Mac), glob system calls can be quite slow. This pull requests caches the same glob calls and returns a cached result if possible. In my testing this reduced the glob calls on the category listings from ~8K to ~1.6K.

Blackfire comparison:

<img width="544" alt="Screenshot 2021-09-06 at 09 51 21" src="https://user-images.githubusercontent.com/2959888/132180261-047d2cb6-e2bd-43ae-afac-02a18b09b4e7.png">




### Related Pull Requests
None

### Fixed Issues (if relevant)
None
1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)
1. Profile any page before this diff is applied
2. Apply diff and reprofile
3. See a reduction in `glob()` calls

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#34024: Greatly improve performance by caching subsequent glob calls

### Resolved issues:
1. [x] resolves magento/magento2#34025: Greatly improve performance by caching subsequent glob calls
